rust/src/rest.rs

```diff
@@ -0,0 +1,114 @@
+use crate::service::Handle;
+use crate::ParticipantPublicKey;
+use crate::crypto::ByteObject;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use bytes::Bytes;
+use bytes::Buf;
+use warp::Filter;
+use warp::reject::Rejection;
+use std::convert::Infallible;
+use warp::http::{StatusCode, Response};
+
+async fn handle_whatever(name: String) -> Result<impl warp::Reply, Rejection> {
+    match name.len() {
+        2 => Ok(warp::reply()),
+        _default => Err(warp::reject::not_found()),
+    }
+
+}
+
+pub async fn serve(addr: impl Into<SocketAddr> + 'static, handle: Handle) {
+    let route = warp::path!("hello" / String)
+//      .map(|name| format!("Hello, {}!", name));
+        .and_then(handle_whatever);
+
+    let message = warp::path!("message")
+        .and(warp::post())
+        .and(warp::body::bytes())
+        .and(with_hdl(handle.clone()))
```

Comment:
so the `.clone()` here is needed to pass `handle` to several of these warp filters (`message` as above, but also `sum_dict`, ...), and then in `with_hdl` a `.clone()` is required by `warp::Filter::map` which expects a `Fn` - without it, the handle moves into the closure, making it a `FnOnce`.

Reply:
handles are just an `Arc`, which are made to be cloned, so I think we should not worry too much about it. clippy sometimes warn about useless clones.

Reply:
thanks for the explanation! also makes more sense now since i'm getting into `warp`. we might be able to reference the handle to conform to the trait bound, but i agree that because of the cheap cloning of arcs is not worth to endeavor into cumbersome borrow checker adventures.